deployment/src/main/java/com/datastax/oss/quarkus/deployment/internal/CassandraClientProcessor.java

```diff
@@ -405,9 +405,21 @@ RuntimeInitializedClassBuildItem runtimeNative() {
   }
 
   @BuildStep
-  ServiceProviderBuildItem registerMutinyResultProducerService() {
-    return new ServiceProviderBuildItem(
-        "com.datastax.oss.driver.api.mapper.result.MapperResultProducerService",
-        MutinyResultProducerService.class.getName());
+  void registerMutinyResultProducerService(
+      BuildProducer<ServiceProviderBuildItem> serviceProviders) {
+    final String mapperResultProducerServiceClassName =
+        "com.datastax.oss.driver.api.mapper.result.MapperResultProducerService";
+    try {
+      Class.forName(
+          mapperResultProducerServiceClassName,
+          false,
+          Thread.currentThread().getContextClassLoader());
+      /* Only register the service provider if com.datastax.oss:java-driver-mapper-runtime
+       * is in the class path */
+      serviceProviders.produce(
+          new ServiceProviderBuildItem(
+              mapperResultProducerServiceClassName, MutinyResultProducerService.class.getName()));
+    } catch (ClassNotFoundException ignored) {
+    }
```

Comment:
First of all, thank you for spotting this :-)

`MutinyResultProducerService` is always present since it's in the runtime module. I think the real problem here is that it implements an interface that may not be present, since indeed the driver mapper artifact is optional.

So I wonder if we couldn't simplify your change to just: 

```suggestion
  ServiceProviderBuildItem registerMutinyResultProducerService() {
    return new ServiceProviderBuildItem(
        "com.datastax.oss.driver.api.mapper.result.MapperResultProducerService",
        "com.datastax.oss.quarkus.runtime.internal.mapper.MutinyResultProducerService");
  }
```

Could you please try it out?

Reply:
I confirm that my suggestion works. I will get an rc2 out today. Thanks again for helping us!